src/login/login.component.tsx

```diff
@@ -5,18 +5,33 @@ import { performLogin } from "./login.resource";
 export default function Login(props: LoginProps) {
   const [username, setUsername] = React.useState("");
   const [password, setPassword] = React.useState("");
+  const [authenticated, setAuthenticated] = React.useState(null);
+  const [errorMessage, setErrorMessage] = React.useState("");
   const [isLoggingIn, setIsLoggingIn] = React.useState(false);
   const [showPassword, setShowPassword] = React.useState(false);
   const passwordInputRef = React.useRef<HTMLInputElement>(null);
   const usernameInputRef = React.useRef<HTMLInputElement>(null);
+  const formRef = React.useRef<HTMLFormElement>(null);
 
   React.useEffect(() => {
     if (isLoggingIn) {
       performLogin(username, password)
         .then(data => {
-          props.history.push("/patient-search");
+          // workaround for now, @openmrs/esm-api response.json() not working!
```

Comment:
What's wrong with @openmrs/esm-api response.json()? It requires doing `data.data`?

Reply:
it actually returns an object of type Response

Reply:
and not json

Reply:
hmm yeah that's a bug let's fix it.

Reply:
Actually this is not a bug -- `response.data` has the json right there and it works totally fine. That is expected behavior.